Objects/memoryobject.c

```diff
@@ -3244,8 +3240,19 @@ memory_iter(PyObject *seq)
     }
 
     it->it_index = 0;
-    Py_INCREF(seq);
     it->it_seq = (PyMemoryViewObject *)seq;
+    Py_INCREF(seq);
+
+    int ndims = it->it_seq->view.ndim;
+    if (ndims == 0) {
+        PyErr_SetString(PyExc_TypeError, "invalid indexing of 0-dim memory");
+        return NULL;
+    }
+    if (ndims != 1) {
+        PyErr_SetString(PyExc_NotImplementedError,
+            "multi-dimensional sub-views are not implemented");
+        return NULL;
+    }
```

Comment:
The error checks should go before you allocate the iteration object. And put the incref back where it was.

Reply:
I pushed a new commit. Does it properly address your comment?